File: app/src/main/java/com/recursia/popularmovies/Screens.java
package com.recursia.popularmovies;

import android.support.v4.app.Fragment;

import com.recursia.popularmovies.presentation.views.fragments.DetailScreenFragment;
import com.recursia.popularmovies.presentation.views.fragments.FavoriteScreenFragment;
import com.recursia.popularmovies.presentation.views.fragments.IntroScreenFragment;
import com.recursia.popularmovies.presentation.views.fragments.MainScreenFragment;
import com.recursia.popularmovies.presentation.views.fragments.PhotoScreenFragment;
import com.recursia.popularmovies.presentation.views.fragments.PopularScreenFragment;
import com.recursia.popularmovies.presentation.views.fragments.SearchScreenFragment;

import ru.terrakok.cicerone.android.support.SupportAppScreen;

public class Screens {
    public static final class DetailScreen extends SupportAppScreen {
        private final int movieId;

        public DetailScreen(int movieId) {
            this.movieId = movieId;
        }

        @Override
        public Fragment getFragment() {
            return DetailScreenFragment.getInstance(movieId);
        }
    }

    public static final class FavoriteScreen extends SupportAppScreen {
        @Override
        public Fragment getFragment() {
            return FavoriteScreenFragment.getInstance();
        }
    }

    public static final class IntroScreen extends SupportAppScreen {
        @Override
        public Fragment getFragment() {
            return IntroScreenFragment.getInstance();
        }
    }

    public static final class PopularScreen extends SupportAppScreen {
        @Override
        public Fragment getFragment() {
            return PopularScreenFragment.getInstance();
        }
    }

    public static final class PhotoScreen extends SupportAppScreen {
        private final String imagePath;

        public PhotoScreen(String imagePath) {
            this.imagePath = imagePath;
        }

        @Override
        public Fragment getFragment() {
            return PhotoScreenFragment.getInstance(imagePath);
        }
    }

    public static final class SearchScreen extends SupportAppScreen {
        @Override
        public Fragment getFragment() {
            return SearchScreenFragment.getInstance();
        }
    }

    public static final class MainScreen extends SupportAppScreen {
        @Override
        public Fragment getFragment() {
            return MainScreenFragment.getInstance();
        }
    }

}

File: app/src/main/java/com/recursia/popularmovies/Screens.kt
package com.recursia.popularmovies

import androidx.fragment.app.Fragment
import com.recursia.popularmovies.presentation.views.fragments.*
import ru.terrakok.cicerone.android.support.SupportAppScreen

class Screens {
    class DetailScreen(private val movieId: Int) : SupportAppScreen() {
        override fun getFragment(): Fragment {
            return DetailScreenFragment.getInstance(movieId)
        }
    }

    class MainScreen : SupportAppScreen() {
        override fun getFragment(): Fragment {
            return MainScreenFragment.getInstance()
        }
    }

    class WelcomeScreen : SupportAppScreen() {
        override fun getFragment(): Fragment {
            return WelcomeScreenFragment.getInstance()
        }
    }

    class AccountScreen : SupportAppScreen() {
        override fun getFragment(): Fragment {
            return AccountScreenFragment.getInstance()
        }
    }

    class ResetPasswordScreen : SupportAppScreen() {
        override fun getFragment(): Fragment {
            return ResetPasswordScreenFragment.getInstance()
        }
    }

    class AuthScreen : SupportAppScreen() {
        override fun getFragment(): Fragment {
            return AuthScreenFragment.getInstance()
        }
    }

    class IntroScreen : SupportAppScreen() {
        override fun getFragment(): Fragment {
            return IntroScreenFragment.instance
        }
    }

    class PhotoScreen(private val imagePath: String) : SupportAppScreen() {
        override fun getFragment(): Fragment {
            return PhotoScreenFragment.getInstance(imagePath)
        }
    }

    class SearchScreen : SupportAppScreen() {
        override fun getFragment(): Fragment {
            return SearchScreenFragment.instance
        }
    }
}
